File: TopInterview150/274_h_index/hIndex.py
class Solution:
    def hIndex(self, citations: List[int]) -> int:
        citations.sort()
        n = len(citations)

        mx = 0
        for i in range(n):
            m = n - citations.index(citations[i], i)
            mx = max(min(m, citations[i]), mx)

        return mx

File: TopInterview150/274_h_index/hIndex.java
class Solution {
    public int hIndex(int[] citations) {
        Arrays.sort(citations);

        int h = 0;
        int n = citations.length;

        for (int i = 0; i < n; i++) {
            int m = n - getIndexOf(citations, citations[i], i);
            h = Math.max(h, Math.min(citations[i], m));
        }

        return h;
    }

    private int getIndexOf(int[] arr, int element, int start) {
        for (int i = start; i < arr.length; i++) {
            if (arr[i] == element)
                return i;
        }

        return -1;
    }
}